api/Silk.Api.Domain/Feature/Users/AddUser.cs

```diff
@@ -0,0 +1,40 @@
+﻿using System;
+using System.Threading;
+using System.Threading.Tasks;
+using MediatR;
+using Silk.Api.Data;
+using Silk.Api.Data.Entities;
+
+namespace Silk.Api.Domain.Feature.Users
+{
+	public static class AddUser
+	{
+		public sealed record Request(string Id, string ApiKey) : IRequest<ApiUser>;
+		
+		public sealed class Handler : IRequestHandler<Request, ApiUser>
+		{
+			private readonly ApiContext _db;
+			public Handler(ApiContext db) => _db = db;
+
+			public async Task<ApiUser> Handle(Request request, CancellationToken cancellationToken)
+			{
+				var user = new ApiUser
+				{
+					ApiKey = new()
+					{
+						GeneratedAt = DateTime.Now,
+						DiscordUserId = request.Id
+					},
+					DiscordId = request.Id,
+					ApiKeyGenerationTimestamp = DateTime.Now,
```

Comment:
Don't need to change anything, just a thought with the ApiKey timestamps. Since the timestamp is generated on the ApiKey itself, syncing it between ApiUser could introduce inconsistencies if not updated

Reply:
It's another thing that was taken out of Discord's book; they only really check token generation timestamps instead of checking whether a token has been invaldiated